crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -620,6 +620,79 @@ mod private {
             Ok(Self { room: room_id, store, events, waited_for_initial_prev_token: false })
         }
 
+        /// Load more events backwards if the last chunk is **not** a gap.
+        ///
+        /// Return `Ok(Some((events, reached_start), _))` if events have been
+        /// inserted, `Ok(None)` if a gap has been inserted or if the
+        /// store is disabled.
+        #[must_use = "Updates as `VectorDiff` must probably be propagated via `RoomEventCacheUpdate`"]
+        pub(in super::super) async fn load_more_events_backwards(
+            &mut self,
+        ) -> Result<
+            Option<((Vec<TimelineEvent>, bool), Vec<VectorDiff<TimelineEvent>>)>,
+            EventCacheError,
+        > {
+            // ⚠️ Propagate all pending updates before we do anything.
+            self.propagate_changes().await?;
+
+            let Some(store) = self.store.get() else {
+                // No store: no events to insert.
+                return Ok(None);
+            };
+
+            let first_chunk = self
+                .events
+                .chunks()
+                .next()
+                // SAFETY: A `LinkedChunk` is never empty, it always contains at least one chunk.
+                .expect("The `LinkedChunk` in `RoomEvents` cannot be empty");
```

Comment:
Isn't there a `first_chunk()` method on the links or something?

Reply:
No, `Ends` (`LinkedChunk::links`) is private. I can add a public method to get the first chunk if you want?